server/src/main/java/org/elasticsearch/search/fetch/subphase/FetchIgnoredMetadataPhase.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.search.fetch.subphase;
+
+import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.index.SortedSetDocValues;
+import org.elasticsearch.common.document.DocumentField;
+import org.elasticsearch.index.mapper.IgnoredFieldMapper;
+import org.elasticsearch.search.fetch.FetchContext;
+import org.elasticsearch.search.fetch.FetchSubPhase;
+import org.elasticsearch.search.fetch.FetchSubPhaseProcessor;
+import org.elasticsearch.search.fetch.StoredFieldsContext;
+import org.elasticsearch.search.fetch.StoredFieldsSpec;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+public class FetchIgnoredMetadataPhase implements FetchSubPhase {
+    @Override
+    public FetchSubPhaseProcessor getProcessor(FetchContext fetchContext) throws IOException {
+        StoredFieldsContext storedFieldsContext = fetchContext.storedFieldsContext();
+        if (storedFieldsContext == null || storedFieldsContext.fetchFields() == false) {
```

Comment:
can you clarify why you need to check storedFieldContext here?

Reply:
So this was a stored field before being a doc values field and this check is necessary to make sure that we do not include it when fetching store fields are disabled. If I understand correctly we have a way to disable fetching stored field and this makes sure we keep the same behaviour.

Reply:
ok thanks for explaining. Disabling fetching of stored fields affects metadata fields that are stored, but not `_index`. This seems to me like a strange feature. I would probably rather always return metadata fields. With that said, I would adapt the test in `SearchHitsTests` instead, and accept that `stored: _none_` will no longer affect `_ignored` when it is no longer stored. That seems consistent with the current behaviour, although it leaks the implementation details, but that is kind of the pitfall of `stored: _none_` anyways. I think it would be worse to pretend that `_ignored` is still stored for the sake of not changing behaviour.

Reply:
This way, anyway, users might not control whether the _ignored field is returned or not anymore right? Considering that is an array of strings might, at least, affect a bit serialisation size of the response for users which are not interested and potentially affect latency when fetching many documents?